### ISSUE TITLE
Sync TextField with UXPin

### ISSUE THREAD

**Issue by scurker** [dequelabs/cauldron#1070]
Some minor UX changes needed:

- Disabled border color should match UXPin
- Focus ring should be 1px smaller (currently appears to be 3px)
- Error field should have a 2px border in its resting state
- Dark theme `field-border-color` should be #74818B

Do we want to get rid of the bottom-border color difference and make it a single solid color that passes WCAG 2.2 AA? See screenshot below:

![Screenshot 2023-06-22 at 12 44 20 PM](https://github.com/dequelabs/cauldron/assets/28606186/285f11dc-3a1a-4732-9a6c-0faeae9ed912)
